### PR TITLE
fix(swap): ensure long running cross chain swaps eventually shows completed status

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR fixes a bug where pending cross chains swaps remain pending forever. The root cause is the query helper reconciles already received transactions with newly received transactions by preferring the already received transaction. This means that previously received pending swaps are not updated when the same transactions is received again with the completed status.

This is a bug that affects not only cross chain swaps (we should always prefer the newest information in case other fields in the transaction have changed.

### Test plan

Cross chains swaps can be settled correctly

### Related issues

n/a

### Backwards compatibility

Y

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [ ] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/transactions/feed/queryHelper.ts [hunk 1]
```diff
@@ -49,18 +49,24 @@ const TAG = 'transactions/feed/queryHelper'
...
   existingTxs: TokenTransaction[],
   incomingTxs: TokenTransaction[]
 ) => {
-  const currentHashes = new Set(existingTxs.map((tx) => tx.transactionHash))
-  const transactionsWithoutDuplicatedHash = existingTxs.concat(
-    incomingTxs.filter((tx) => !isEmpty(tx) && !currentHashes.has(tx.transactionHash))
-  )
-  transactionsWithoutDuplicatedHash.sort((a, b) => {
+  const transactionsByTxHash: { [txHash: string]: TokenTransaction } = {}
+  existingTxs.forEach((transaction) => {
```
Comment (jophish) [inline src/transactions/feed/queryHelper.ts:60]: nit: can do 
```ts
[...existingTxs, ...incomingTxs].forEach((transaction) => {
    transactionsByTxHash[transaction.transactionHash] = transaction
  })
```
to save some space

File: src/transactions/reducer.ts [hunk 2]
```diff
@@ -239,10 +239,10 @@ export const pendingStandbyTransactionsSelector = createSelector(
     return transactions
       .filter((transaction) => transaction.status === TransactionStatus.Pending)
       .map((transaction) => ({
-        ...transaction,
         transactionHash: transaction.transactionHash || '',
         block: '',
         fees: [],
+        ...transaction, // in case the transaction already has the above (e.g. cross chain swaps), use the real values
```
[the remark this answers is not in the document]
Comment (jophish) [reply src/transactions/reducer.ts:245]: can we cover this new behavior with a test to ensure new TX data overrides old data?

**Review by jophish — APPROVED** [review]
LGTM, just small comments

File: src/transactions/feed/queryHelper.ts (hunk 1 above)
Comment (kathaypacific) [reply src/transactions/feed/queryHelper.ts]: TY, updated!

**Pull request merged**